src/main/scala/net/ripe/rpki/publicationserver/repository/DataFlusher.scala

```diff
@@ -0,0 +1,245 @@
+
+  def snapshotStream(sessionId: String, serial: Long): HashingSizedStream =
+    new HashingSizedStream(rrdpFileStream(sessionId, serial, Rrdp.snapshotFilename))
+
+  private def rrdpFileStream(sessionId: String, serial: Long, localName: String) = {
+    val stateDir = Files.createDirectories(Paths.get(conf.rrdpRepositoryPath, sessionId, String.valueOf(serial)))
+    new FileOutputStream(stateDir.resolve(localName).toFile)
+  }
+
+  protected def writeLogEntryToRsync(operation: String, uri: URI, bytes: Option[Bytes]) =
+    (operation, bytes) match {
+      case ("INS", Some(b)) => rsyncWriter.writeFile(uri, b)
+      case ("UPD", Some(b)) => rsyncWriter.writeFile(uri, b)
+      case ("DEL", _)       => rsyncWriter.removeFile(uri)
+      case anythingElse =>
+        logger.error(s"Log contains invalid row $anythingElse")
+    }
+
+  private def writeObjectToSnapshotFile(uri: URI, bytes: Bytes, stream: HashingSizedStream): Unit =
+    writePublish(uri, bytes, stream)
+
+  def writeLogEntryToDeltaFile(operation: String, uri: URI, oldHash: Option[Hash], bytes: Option[Bytes], stream: HashingSizedStream) = {
+    (operation, oldHash, bytes) match {
+      case ("INS", None, Some(bytes)) =>
+        writePublish(uri, bytes, stream)
+      case ("UPD", Some(Hash(hash)), Some(bytes)) =>
+        IOStream.string(s"""<publish uri="${attr(uri.toASCIIString)}" hash="$hash">""", stream)
+        IOStream.string(Bytes.toBase64(bytes).value, stream)
+        IOStream.string("</publish>\n", stream)
+      case ("DEL", Some(Hash(hash)), None) =>
+        IOStream.string(s"""<withdraw uri="${attr(uri.toASCIIString)}" hash="$hash"/>\n""", stream)
+      case anythingElse =>
+        logger.error(s"Log contains invalid row $anythingElse")
+    }
+  }
+
+  private def writePublish(uri: URI, bytes: Bytes, stream: HashingSizedStream) = {
+    IOStream.string(s"""<publish uri="${attr(uri.toASCIIString)}">""", stream)
+    IOStream.string(Bytes.toBase64(bytes).value, stream)
+    IOStream.string("</publish>\n", stream)
+  }
+
+  def withOS(createStream: => HashingSizedStream)(f : HashingSizedStream => Unit) = {
+    val stream = createStream
+    try {
+      f(stream)
+      stream.summary
+    } finally {
+      stream.close()
+    }
+  }
+
+  def afterRetainPeriod = new Date(System.currentTimeMillis() + conf.unpublishedFileRetainPeriod.toMillis)
+
+  def initialRrdpRepositoryCleanup(sessionId: UUID)(implicit session: DBSession) = {
+
+    val now = Instant.now()
+    val oldEnough = FileTime.from(now.minus(conf.unpublishedFileRetainPeriod.toMillis, ChronoUnit.MILLIS))
+
+    // Delete files related to the current sessions
+    updateRrdpRepositoryCleanup()
+
+    // Cleanup files that are left from some previously existing sessions
+    //
+    // First remove the ones that are more than time T old
+    logger.info(s"Removing all the sessions in RRDP repository except for $sessionId")
```

Comment:
Message seems to mismatch the code?

Reply:
I renamed the function so now it all matches, I think.